Stack/Core/Stack/State/NodeState.cs

```diff
@@ -2369,23 +2359,33 @@ public virtual void RemoveNotifier(ISystemContext context, NodeState target, boo
         {
             NodeState nodeState = null;
 
-            lock (this.m_notifiersLock)
+            lock (m_notifiersLock)
             {
                 if (m_notifiers != null)
                 {
-                    for (int ii = 0; ii < m_notifiers.Count; ii++)
+                  
+                    Notifier[] notifiers = null;
+                    for (int ii = 0; ii < m_notifiers.Length; ii++)
                     {
                         Notifier entry = m_notifiers[ii];
 
                         if (Object.ReferenceEquals(entry.Node, target))
                         {
                             nodeState = entry.Node;
-                            m_notifiers.RemoveAt(ii);
+
+                            notifiers = new Notifier[m_notifiers.Length - 1];
+                            //copy every element to notifiers except at index ii
+                            Array.Copy(m_notifiers, 0, notifiers, 0, ii);
+                            Array.Copy(m_notifiers, (ii + 1), notifiers, ii, (m_notifiers.Length - ii - 1));
```

Comment:
Add check that we are not removing from the end when this is unnecessary
Hint: http://stackoverflow.com/a/457501  :)